modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/utils/sortItems.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {IOrderable} from './types';
+
+/**
+ * sorts the provided items array according to the itemsOrder comma-separated list of ids.
+ * If array contains items not included in the list of ids, then those are appended after
+ * Example:
+ * 		items = [ {id: 1}, {id: 4}, {id: 2}, {id: 3} ]
+ * 		itemsOrder = "2, 3, 1"
+ * 		output is [ {id: 2}, {id: 3}, {id: 1}, {id: 4} ]
+ * Optionally, not included items can be sorted by creation date
+ *
+ * @param items {IOrderable[]}
+ * @param itemsOrder {string}
+ * @param useCreationDate {boolean}
+ * @returns {Array}
+ */
+export default function sortItems(
+	items: IOrderable[],
+	itemsOrder: string,
+	useCreationDate: boolean = false
+): IOrderable[] {
+	const itemsOrderArray = itemsOrder?.split(',') || ([] as string[]);
+
+	let included: IOrderable[] = [];
+	let notIncluded: IOrderable[] = [];
+
+	included = itemsOrderArray
+		.map((itemId) => items.find((item) => item.id === Number(itemId)))
+		.filter(Boolean) as IOrderable[];
+
+	notIncluded = items.filter(
+		(item) => !itemsOrderArray.includes(String(item.id))
+	);
+
+	if (useCreationDate) {
+		const notIncludedIdsMap = {} as {[key: number]: number};
+
+		notIncluded.forEach((item) => {
+			notIncludedIdsMap[item.id] = Date.parse(item.dateCreated);
+		});
+
+		notIncluded = notIncluded.sort((item1, item2) =>
+			notIncludedIdsMap[item1.id] > notIncludedIdsMap[item2.id]
+				? 1
+				: notIncludedIdsMap[item1.id] < notIncludedIdsMap[item2.id]
+				? -1
+				: 0
+		);
+	}
```

Comment:
Could we simplify this to something like 
```js
notIncluded = notIncluded.sort((item1, item2) =>
    Date.parse(item1.dateCreated) - Date.parse(item2.dateCreated)
);
```

Reply:
Main reason I separated this is to avoid repetitive Date.parse calls, but concept you shown can also be used by just substrracting `notIncludedIdsMap[item1.id] - notIncludedIdsMap[item2.id]` so I'll change it

Reply:
Nice consideration! Just a non-blocking nit here, it's not very clear from the name `notIncludedIdsMap` what it contains. Maybe `creationDates`?